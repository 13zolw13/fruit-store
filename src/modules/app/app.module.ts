import { ApolloDriver, ApolloDriverConfig } from '@nestjs/apollo';
import { Module } from '@nestjs/common';
import { GraphQLModule } from '@nestjs/graphql';
import { join } from 'path';
import { DatabaseModule } from '../database/database.module';
import { ProductsModule } from '../products/products.module';
import { AppController } from './app.controller';
import { AppService } from './app.service';

@Module({
  imports: [
    GraphQLModule.forRoot<ApolloDriverConfig>({
      driver: ApolloDriver,
      debug: true,
      playground: false,
      autoSchemaFile: join(process.cwd(), 'src/schema.gql'),
      sortSchema: true,
    }),
    ProductsModule,
    DatabaseModule,
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
