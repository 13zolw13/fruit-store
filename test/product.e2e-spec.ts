import { INestApplication } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import { AppModule } from '../src/modules/app/app.module';
import request = require('supertest');

describe('AppController (e2e)', () => {
  let app: INestApplication;
  const gql = '/graphql';
  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  afterAll(async () => {
    await app.close();
  });

  describe('Create Product', () => {
    it('should return created product', async () => {
      return request(app.getHttpServer())
        .post(gql)
        .send({
          query: `mutation{
  	createProduct(createProductInput:{
      name:"bob"
    }){
    name
    id
  }
}`,
        })
        .expect(200)
        .expect((res) => {
          expect(res.body.data.createProduct.name).toBe('bob');
          expect(res.body.data.createProduct.id).toBeDefined();
        });
    });
  });

  describe('Find all products', () => {
    it('should return list of products', async () => {
      return request(app.getHttpServer())
        .post(gql)
        .send({
          query: `query{ products{ name, price}}`,
        })
        .expect(200)
        .expect((res) => {
          expect(res.body.data.products).toBeDefined();
          expect(res.body.data.products[0].name).toBe('bob');
          expect(res.body.data.products).toMatchSnapshot();
        });
    });
  });
});
