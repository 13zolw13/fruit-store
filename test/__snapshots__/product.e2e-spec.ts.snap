// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) Create Product should return created product 1`] = `
Array [
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
  Object {
    "name": "bob",
    "price": null,
  },
]
`;
